package arrays;

public class SetMatricesZeroes {
    static void setZeroes(int[][] matrix){
        int m = matrix.length;
        int n = matrix[0].length;

        boolean[] row = new boolean[m];
        boolean[] col = new boolean[n];

        for (int i = 0; i< m; i++){
            for (int j = 0; j< n; j++){
                if (matrix[i][j] == 0) {
                    row[i] = true;
                    col[j] = true;
                }
            }
        }
        

        // m to zero
        for (int i = 0; i< m; i++){
            if (row[i]){
                for (int j = 0; j < n; j++){
                    matrix[i][j] = 0;
                }
            }
        }

        // columns to zero
        for (int j = 0;j< n; j++){
            if (col[j]){
                for (int i = 0; i < m; i++){
                    matrix[i][j] = 0;
                }
            }
        }
        for (int i = 0; i< m; i++){
            for (int j = 0; j< n; j++){
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        int[][] matrix = {
                {1,1,1},
                {1,0,1},
                {1,1,1}
        };
        setZeroes(matrix);

    }
}
